### ISSUE TITLE
Cannot assign TypedDict subtype to dict[Any, Any]

### ISSUE THREAD

**Issue by kyle-query** [microsoft/pyright#6658]
**Describe the bug**

pyright does not allow `TypedDict` subtypes to be assigned to `dict` variables. `TypedDict` values are always represented as `dict` with `str` keys at runtime.

**Code or Screenshots**

```python
from typing import TypedDict, Any

class A(TypedDict):
    a: int

def test(x: dict[Any, Any]):
    return

arg: A = {"a": 1}

test(arg) # "A" is incompatible with "dict[Any, Any]"
```

**VS Code extension or command-line**

This can be reproduced at least as far back as pyright 1.1.333

**Comment by kyle-query** [comment]
In #6643, you mentioned `dict` is a nominal type and `TypedDict` is a structural type. I believe that is the reason why pyright is reporting an error, but I'll reopen this to get confirmation.

**Comment by kyle-query** [comment]
Closing again... I am able to workaround the problem by using `Mapping[str, Any]` as the bound. I suppose this works because `Mapping` is a structural type.

**Comment by erictraut** [comment]
A `TypedDict` is not type compatible with a `dict`, but it is type compatible with a `Mapping`, which you've discovered. That's not because a class created with `TypedDict` is structural. If you're interested in the reasoning, refer to [PEP 589](https://peps.python.org/pep-0589/#type-consistency), which introduced `TypedDict` to the type system.

**Comment by kyle-query** [comment]
Thanks for the reference Eric!
